tests/ui/abi/statics/static-mut-foreign.stderr

```diff
@@ -1,20 +1,90 @@
 warning: creating a shared reference to mutable static is discouraged
-  --> $DIR/static-mut-foreign.rs:31:18
+  --> $DIR/static-mut-foreign.rs:21:16
+   |
+LL |     assert_eq!(rust_dbg_static_mut, 3);
+   |                ^^^^^^^^^^^^^^^^^^^ shared reference to mutable static
+   |
+   = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
+   = note: this will be a hard error in the 2024 edition
+   = note: this shared reference has lifetime `'static`, but if the static ever gets mutated, or a mutable reference is created, then any further use of this shared reference is Undefined Behavior
+   = note: `#[warn(static_mut_refs)]` on by default
+help: use `addr_of!` instead to create a raw pointer
+   |
+LL |     assert_eq!(addr_of!(rust_dbg_static_mut), 3);
```

Comment:
This suggestion is wrong, that is, if applied it would change the semantics of the code. 

Maybe disable suggestions if any of the parts of the `& ...` expression are from different expansion contexts?